Core/Object Arts/Dolphin/MVP/MultiSelectListViewTest.cls

```diff
@@ -17,6 +17,116 @@ initializePresenter
 sortSelections
 	^#(49 50)!
 
+testNewSelectionsClickOutsideListWithModifiers
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1).
+	(OrderedCollection new)
+		add: #(#control);
+		add: #(#shift);
+		add: #(#control #shift);
+		do: 
+				[:modifiers |
+				event := self mouseDownEventOnItem: 0 buttons: (modifiers copyWith: #left).
+				self assert: (presenter newSelectionsFromEvent: event) = #(1)].!
+
+testNewSelectionsCtrlClickOnSelectedItem
+	| event |
+	self setUpForSelectionTesting.
+	presenter selectionsByIndex: #(1 2).
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(2).
+	presenter selectionsByIndex: #(1).
+	self assert: (presenter newSelectionsFromEvent: event) = #().!
+
+testNewSelectionsCtrlClickOnUnselectedItem
+	| event |
+	self setUpForSelectionTesting.
+	event := self mouseDownEventOnItem: 1 buttons: #(#left #control).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1).
+	presenter selectionsByIndex: #(2).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1 2).!
+
+testNewSelectionsCtrlShiftClickAdditive
+	| event |
+	self setUpForSelectionTesting.
+	presenter
+		selectionsByIndex: #(1 3);
+		selectionMarkIndex: 3.
+	event := self mouseDownEventOnItem: 7 buttons: #(#left #control #shift).
+	self assert: (presenter newSelectionsFromEvent: event) = #(1 3 4 5 6 7).
```

Comment:
This is a good test as this behaviour only applies if Ctrl is down. Shift only removes the selection of 1. Do you have a test of the shift only behaviour of the result being selection between the mark and clicked item? Maybe I just haven't spotted that yet.